### ISSUE TITLE
`parse_many` silently discards invalid sections

### ISSUE THREAD

**Issue by mhils** [jcreekmore/pem-rs#27]
First off, thanks for the super helpful project! :cake:

I was looking at exposing `pem::parse_many` to Python users via the `cryptography` package (https://github.com/pyca/cryptography/issues/6351#issuecomment-932148321). In that process I discovered that `parse_many` silently discards all invalid sections, which I found quite surprising. That makes it hard to provide good error messages as I cannot differentiate between malformed and missing sections.

Long story short, would you entertain the idea of either changing `parse_many` to return a `Result<Vec<Pem>>` or adding an alternative method name (`parse_all`, `parse_many_strict`?) that returns an error on the first parsing failure? I'd be happy to send a PR with whatever you prefer. 😃 

For reference, here's the current `parse_many` implementation: 

https://github.com/jcreekmore/pem-rs/blob/b2cfec184815fecace9b5b1844eb6b15bda69793/src/lib.rs#L320-L326

**Comment by jcreekmore** [comment]
I am fine with updating `parse_many` to return a `Result<Vec<Pem>>`. If you want to send a PR with that, I will take a look and probably look at releasing a 1.0 version of the package with that change.

**Comment by mhils** [comment]
This is fixed in 1.0, thank you for the super quick turnaround. 🍰 🦀
